src/shogun/classifier/GaussianProcessClassification.cpp

```diff
@@ -139,26 +138,32 @@ bool GaussianProcessClassification::train_machine(std::shared_ptr<Features> data
 	// check whether given combination of inference method and likelihood
 	// function supports classification
 	require(m_method, "Inference method should not be NULL");
-	auto lik=m_method->get_model();
-	require(m_method->supports_binary() || m_method->supports_multiclass(), "{} with {} doesn't support "
-			"classification\n", m_method->get_name(), lik->get_name());
-
+	random_seed_callback(m_method->get<LikelihoodModel>("likelihood_model").get());
```

Comment:
not sure it is a good idea to have this in here? What if a user gave a different seed to the likelihood?

Reply:
maybe we should use the previous way? just pass the seed to the likelihood, and don't' make the GP seedable.

Reply:
I like the GP being seedable, but I would expect:

* if setting the seed to GP, the seed is propagated to all yet set members
* if a user decides to set the seed of a member (after it has been set), that is the seed the object should have

* If the user sets the seed for the GP, then adds a member, and then runs the method, the seed is not expected to be passed on the the member